pkg/chunk/schema_config.go

```diff
@@ -148,6 +151,16 @@ func (cfg *SchemaConfig) ForEachAfter(t model.Time, f func(config *PeriodConfig)
 func (cfg PeriodConfig) CreateSchema() (BaseSchema, error) {
 	buckets, bucketsPeriod := cfg.createBucketsFunc()
 
+	// Ensure Index table prefix exists
```

Comment:
We have `periodCfg.validate()` that should have the validation logic for the `PeriodConfig`. Is there any blocker to move this logic there?

Reply:
The `periodCfg.validate()` ultimately calls `[createSchema](https://github.com/cortexproject/cortex/blob/master/pkg/chunk/schema_config.go#L148)` function which validates the Period config for index and chunk. It validates by creating schema and fails if it fails. That's why I have put the check statement here.